tools/stronghold/src/api/compatibility.py

```diff
@@ -225,3 +244,75 @@ def _check_variadic_parameters(
         yield api.violations.VarArgsDeleted(func=func, line=after.line)
     if before.variadic_kwargs and not after.variadic_kwargs:
         yield api.violations.KwArgsDeleted(func, line=after.line)
+
+
+def _check_type_compatibility(
+    type_before: api.types.TypeHint, type_after: api.types.TypeHint
+) -> bool:
+    """Checks that the type annotations are compatible.
+    Returns True if compatible.
+    """
+    # If annotations are identical, they are compatible
+    if type_before == type_after:
+        return True
+
+    # If either of the annotations is None, then we can't make a compatibility judgement
+    # because Python allows functions to have untyped parameters.
+    if type_before is None or type_after is None:
```

Comment:
That's a good idea, I can add it in a separate PR, since the current BC linter implementation doesn't distinguish between warning types (everything is an error and log level depends only on the suppression).